src/lib/ui/view/Todo.svelte

```diff
@@ -21,6 +23,39 @@
 
 	$: shouldLoadEntities = browser && $socket.open;
 	$: entities = shouldLoadEntities ? dispatch('QueryEntities', {space_id: $space.space_id}) : null;
+	$: tiesResult = shouldLoadEntities ? dispatch('ReadTies', {space_id: $space.space_id}) : null;
+	let ties: Tie[] | undefined;
+	//TODO move this call to the UI to get arch & caching
+	// eslint-disable-next-line @typescript-eslint/no-floating-promises
+	$: tiesResult?.then((data) => {
+		if (data.ok) {
+			ties = data.value.ties;
+		}
+	});
+
+	$: itemsByEntity = $entities && ties ? toItemsByEntity($entities, ties) : null;
+
+	let entityById: Map<number, Readable<Entity>> | null = null;
+	$: entityById = $entities && new Map($entities.map((e) => [get(e).entity_id, e]));
+
+	//TODO do caching here
+	const toItemsByEntity = (
+		entities: Array<Readable<Entity>>,
+		ties: Tie[],
+	): Map<Readable<Entity>, Array<Readable<Entity>>> => {
+		const map: Map<Readable<Entity>, Array<Readable<Entity>>> = new Map();
+		for (const tie of ties) {
+			if (tie.type !== 'HasItem') continue;
+			const source = entities.find((e) => get(e).entity_id === tie.source_id)!;
+			const dest = entities.find((e) => get(e).entity_id === tie.dest_id)!;
+			let items = map.get(source);
+			if (!items) {
+				map.set(source, (items = []));
+			}
+			items.push(dest);
```

Comment:
whoops I sidetracked us in conversation, this is the point I dropped:

```suggestion
			if (!items.includes(dest)) items.push(dest);
```